modules/nf-core/summarizedexperiment/summarizedexperiment/templates/summarizedexperiment.r

```diff
@@ -155,24 +155,43 @@ if ('assay_names' %in% names(args_opt)){
 
 # Build and verify the main assays list for the summarisedexperiment
 
-assay_list <- lapply(matrix_files, function(m){
-    mat <- read_delim_flexible(m, row.names = 1, stringsAsFactors = FALSE)
-    mat[,sapply(mat, is.numeric), drop = FALSE]
-})
+assays_list <- list()
+for(file in matrix_files) {
+    mat <- read_delim_flexible(file, row.names = 1, stringsAsFactors = FALSE)
+    mat <- mat[, sapply(mat, is.numeric), drop = FALSE]
+
+    if(grepl("gene_tpm", file)) {
+        assays_list[["tpm"]] <- as.matrix(mat)
+    } else if(grepl("gene_counts_length_scaled", file)) {
+        assays_list[["counts_length_scaled"]] <- as.matrix(mat)
+    } else if(grepl("gene_counts_scaled", file)) {
+        assays_list[["counts_scaled"]] <- as.matrix(mat)
+    } else if(grepl("gene_counts", file)) {
+        assays_list[["counts"]] <- as.matrix(mat)
+    } else if(grepl("gene_lengths", file)) {
+        assays_list[["gene_lengths"]] <- as.matrix(mat)
+    } else if(grepl("transcript_tpm", file)) {
+        assays_list[["tpm"]] <- as.matrix(mat)
+    } else if(grepl("transcript_counts", file)) {
+        assays_list[["counts"]] <- as.matrix(mat)
+    } else if(grepl("transcript_lengths", file)) {
+        assays_list[["lengths"]] <- as.matrix(mat)
+    }
+}
```

Comment:
So far, the module has been quite generic, while with this change it would become rather specific to the RNAseq pipeline. Would it be possible to pass a mapping to the module as an input channel, something like 

```groovy
SUMMARIZEDEXPERIMENT_SUMMARIZEDEXPERIMENT(
     matrix_files, 
     col_data,
     row_data,
     [
         "gene_counts_length_scaled.tsv" : "tpm",
        "gene_counts_scaled.tsv": "counts_scaled",
        ...
     ]
)
```

This would also allow us to be fully backwards-compatible: If no such map is provided, the behavior doesn't change. If it is provided, only the matrices specified in the map are used and stored in the appropriate slot.

Reply:
I think this is getting overly complicated. Why don't we just add a param to the module for the assay name, and then create a separate module to merge the objects? It'd only be about 2 lines of R.

Reply:
You mean basically that SummarizedExperiment module takes exactly one matrix and then you merge them in a separate process? 
There are some downsides, but not sure how much of a problem they are in practice:
 * it's less efficient due to duplicated IO
 * the merge function may need additional checks (e.g. colData/rowData is the same across all objects), so it's likely not just 2 lines of R code. 

Ultimately, I'm fine with either solution. Your call!

Reply:
Wait, actually there's an `assay_names` arg for naming the assays: https://github.com/nf-core/modules/blob/fa30cb23a4ad91e509b0c91399ab3b0cc76e858f/modules/nf-core/summarizedexperiment/summarizedexperiment/templates/summarizedexperiment.r#L150. What's the issue with just using that?

Reply:
What I mean is, the new code here should not be necessary. The code will already apply the specified names to the list of input matrix files, which will then be retained when those files are converted to the assays array.